src/lib.rs

```diff
@@ -225,6 +225,22 @@ impl<I: Indexed> Range<I> {
             row_end: row_end,
             col_start: col_start,
             col_end: col_end,
+            length: None
+        }
+    }
+
+    pub fn new_with_len(row_start: Row<I>,
```

Comment:
we should be consistent about naming - use either `len` or `length` for both the field and in the methods.